crates/provider-sqldb-postgres/README.md

```diff
@@ -0,0 +1,181 @@
+# 🐘 SQL Database Postgres Provider
+
+This capability provider implements the [wasmcloud:sqldb-postgres][wasmcloud-sqldb-postgres-wit] WIT package, which enables SQL-driven database interaction with a [Postgres][postgres] database cluster.
+
+This provider handles concurrent component connections, and components can choose how they manage connections:
+
+| Management type    | Description                                                       |
+|--------------------|-------------------------------------------------------------------|
+| Completely Managed | Defined at provider startup via host-supplied named configuration |
+| Link Driven        | Defined via config for each component-provider link               |
+| Component Driven   | Controlled completely by connected components                     |
```

Comment:
I feel like those details can be modified at the link level for a component, correct?

Reply:
So right now what you can do is *pick* the pre-existing profile -- specifying overrides is a new feature that hasn't really been considered up until now, AFAIK (or at least I haven't seen it!)

Then it's a question of *what* we want to override -- only DB name is probably pretty safe, so I can add that if that satisfies your concern here, and allows for people to at least implement some of that multi-tenancy pattern I noted earlier

Reply:
For what it is worth - I prefer the way it is currently. The way I see it is a north star should be a WIT world should be able to manage use cases for each level of 'convention over composition'. For example:

- level 1 - with this interface, it seems to allow components to manage everything, which allows computations to be completely decoupled from the state the computation is executed on.
- level 2 - with this interface, it allows for components to explicitly couple a computation with the state it needs to execute on
- level 3 - with this interface, it allows for components to explicitly couple a computation with the state it needs to execute on, and allows to specify the expected output of the transform (result-rows)

Very happy with this interface!

Reply:
@LUK3ARK So I definitely understand and agree, but only for custom interfaces (or product specific interfaces). What I don't want us to be in the business of is encouraging the component to do the same thing they are doing now: coupling specific connection string types or details in their actual code. Since this is going to be a generic interface that multiple things can implement, I am strongly against putting that style of management in a more generic, community supported contract, but fully supportive of a cloud provider or other third party/service provider having this kind of functionality if they so choose.

Reply:
Well the point that this might tempt people to put credentials in their WASM binaries (which are certainly not obfuscated in any way) is certainly quite bad (at the very least until [a good answer for moving secrets](https://github.com/wasmCloud/wasmCloud/issues/2190) arrives)...

The medium here is probably level 1 and 2 + overrides for profile driven connections (most importantly the DB name) -- we can leave the completely manual connection management unimplemented by this particular provider

Reply:
Just want to note that as always I really appreciate you kicking the tires @LUK3ARK , helping us find bugs and think through corner cases and general design!